app/src/main/java/com/gathering/android/event/putevent/pic/AddPicScreen.kt

```diff
@@ -274,4 +240,61 @@ class AddPicScreen : BottomSheetDialogFragment(), AddPicNavigator {
             ErrorText(error = errorMessage)
         }
     }
+
+    @Composable
+    @OptIn(ExperimentalPermissionsApi::class)
+    private fun CameraAndGalleryButtons(
+        cameraPermissionState: PermissionState,
+        onCameraClick: () -> Unit,
+        onGalleryClick: () -> Unit,
+        galleryPermissionState: PermissionState
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(10.dp),
+            horizontalArrangement = Arrangement.Center
+        ) {
+            Button(onClick = {
+                if (cameraPermissionState.status.isGranted) {
```

Comment:
this logic should be extracted out of UI and live in ViewModel. view state should have a property indicating the permission state. viewmodel also needs a dependency like PermissionManager to handle asking for permission.